### PR TITLE
KAFKA-17846: ClientTelemetryReporter does not log trace-level message

### PR DESCRIPTION
**Pull request by clarkwtc**
Reverse the condition to show log trace-level message

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by clarkwtc** [comment]
@chia7712 @apoorvmittal10 
Could you review the PR, please?
Your feedback would be good to me.

File: clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java [hunk 1]
```diff
@@ -337,33 +336,33 @@ public long timeToNextUpdate(long requestTimeoutMs) {
...
-                        msg = isTraceEnabled ? "" : String.format("the client will wait before submitting the next %s network API request", apiName);
+                        msg = String.format("the client will wait before submitting the next %s network API request", apiName);
                     }
                     break;
                 default:
                     throw new IllegalStateException("Unknown telemetry state: " + localState);
             }
 
-            if (isTraceEnabled) {
+            if (log.isTraceEnabled()) {
```
Comment (apoorvmittal10) [inline clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: Do you think this check is also not adding any value?

**Review by apoorvmittal10 — APPROVED** [review]
LGTM, thanks for fixing this. I have left one comment.

File: clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java (hunk 1 above)
Comment (clarkwtc) [reply clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: Remove this check? 
Always trace the current state of the client telemetry.

Comment (apoorvmittal10) [reply clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: So basically `log.trace` will anyways not log if trace is not enabled.
The check `log.isTraceEnabled()` generally saves us creating extra strings with object defined as params but here we are using `format parameters` for log hence the extra strings will not be created. This check is not adding any value hence we can remove it.

Comment (chia7712) [reply clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: +1 to remove the check, as the trace-level log is cheaper due to the use of placeholders.

Comment (clarkwtc) [reply clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: @apoorvmittal10 You're right
Thank you both for the kind suggestions, I have removed the check.

**Review by apoorvmittal10 — APPROVED** [review]
Thanks for addressing the comment. LGTM!

File: clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java [hunk 2]
```diff
@@ -337,35 +336,33 @@ public long timeToNextUpdate(long requestTimeoutMs) {
...
                     apiName = (localState == ClientTelemetryState.SUBSCRIPTION_NEEDED) ? ApiKeys.GET_TELEMETRY_SUBSCRIPTIONS.name : ApiKeys.PUSH_TELEMETRY.name;
                     long timeRemainingBeforeRequest = localLastRequestMs + localIntervalMs - nowMs;
                     if (timeRemainingBeforeRequest <= 0) {
                         timeMs = 0;
-                        msg = isTraceEnabled ? "" : String.format("the wait time before submitting the next %s network API request has elapsed", apiName);
+                        msg = String.format("the wait time before submitting the next %s network API request has elapsed", apiName);
                     } else {
                         timeMs = timeRemainingBeforeRequest;
-                        msg = isTraceEnabled ? "" : String.format("the client will wait before submitting the next %s network API request", apiName);
+                        msg = String.format("the client will wait before submitting the next %s network API request", apiName);
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: I assumed the new style is to call `log.trace` directly :smile:  for example:
```java
log.trace("For telemetry state {}, returning the value {} ms; the client will wait before submitting the next {} network API request", localState, timeMs, apiName);
```

Comment (apoorvmittal10) [reply clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: Yeah that ll be better else the Strings are created for msg.

Comment (clarkwtc) [reply clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java]: Looks better.
Thank you for this suggestion.
done.

**Review by apoorvmittal10 — APPROVED** [review]

**Review by chia7712 — APPROVED** [review]

**Pull request merged**